joomla/joomla-cms
# [4] Introducing Joomla’s first blank state

### Summary of Changes

So [I mentioned about the possibility of blank states](https://github.com/joomla/joomla-cms/issues/30887#issuecomment-702959686) in Joomla 4 as a comment on someone's PR a long time ago. It received quite a bit of positive feedback but no one has actually attempted it

So here is "an attempt"

The concept is that rather than a new user getting a first time out of the box experience with "no matching results" or a state that is unhelpful, that the user receives a welcome and an introduction to each area. 

This PR is a PROOF OF CONCEPT for Articles ONLY - if it received praise then I might do the others, and others can join in too..

The blank state template is ONLY USED when you have no articles in your database. It is NOT used if a search/filter produces no results (in that case, you would correctly get "No Matching Results" message like you currently do. 

### To be clear

This is not a new concept. Everyone else is already doing it. 

More examples of Empty State Design:
https://www.toptal.com/designers/ux/empty-state-ux-design

### Testing Instructions

Apply PR on a site with no articles. 
Add article - blank state design disappears and you see a list of articles
filter the articles by something (search for kashfdksjhgfkjsdhf) and see that you DONT see the blank slate but you DO get "No Matching Results"
Trash your article -  and see that you DONT see the blank slate
Empty your trash (leaving ZERO articles in the db)  and see that you DO now see the blank slate


### Actual result BEFORE applying this Pull Request

At the moment the out of the box experience (without demo data) is a bunch of left menu items that all lead to blank pages saying "no matching results" 

User is "welcomed" to Joomla with a negative messages

<img width="1603" alt="Screenshot 2021-04-23 at 14 27 04" src="https://user-images.githubusercontent.com/400092/115878073-08f3f500-a440-11eb-9d48-28428ac8e1bb.png">


### Expected result AFTER applying this Pull Request

Empty State Design / Blank state Design pattern is used. 

### Documentation Changes Required

Screenshots. 

<img width="1601" alt="Screenshot 2021-04-23 at 15 15 55" src="https://user-images.githubusercontent.com/400092/115884454-d7325c80-a446-11eb-9eb1-51eab9147c63.png">
